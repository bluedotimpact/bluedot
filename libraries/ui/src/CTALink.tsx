import clsx from 'clsx';
import React from 'react';

// Create a styled link component using CTAButton styling
export const CTALink: React.FC<{
  href: string;
  className?: string;
  variant?: 'primary' | 'secondary';
  withChevron?: boolean;
  children: React.ReactNode;
}> = ({
  href,
  className,
  variant = 'primary',
  withChevron = false,
  children,
}) => {
  return (
    <a
      href={href}
      target="_blank"
      rel="noopener noreferrer"
      className={clsx(
        'cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base',
        {
          'cta-button--primary bg-bluedot-normal text-white hover:bg-bluedot-normal': variant === 'primary',
          'cta-button--secondary bg-transparent text-bluedot-normal border border-bluedot-normal hover:bg-bluedot-lighter': variant === 'secondary',
        },
        'px-4 py-2',
        className,
      )}
    >
      <span className="cta-button__text">{children}</span>
      {withChevron && (
      <span className="cta-button__chevron ml-3">
        <img
          src={variant === 'primary' ? '/icons/chevron_white.svg' : '/icons/chevron_blue.svg'}
          alt="â†’"
          className="cta-button__chevron-icon w-2 h-2"
        />
      </span>
      )}
    </a>
  );
};

export default CTALink;
