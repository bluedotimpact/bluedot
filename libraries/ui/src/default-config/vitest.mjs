/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
