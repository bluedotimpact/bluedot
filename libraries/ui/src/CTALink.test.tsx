import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import { CTALink } from './CTALink';

describe('CTALink', () => {
  test('renders primary variant as expected', () => {
    const { container } = render(
      <CTALink
        href="https://example.com"
        variant="primary"
      >
        Click me
      </CTALink>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders secondary variant as expected', () => {
    const { container } = render(
      <CTALink
        href="https://example.com"
        variant="secondary"
      >
        Click me
      </CTALink>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with chevron as expected', () => {
    const { container } = render(
      <CTALink
        href="https://example.com"
        withChevron
      >
        Click me
      </CTALink>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with custom className', () => {
    const { container } = render(
      <CTALink
        href="https://example.com"
        className="custom-class"
      >
        Click me
      </CTALink>,
    );
    expect(container).toMatchSnapshot();
  });
});
