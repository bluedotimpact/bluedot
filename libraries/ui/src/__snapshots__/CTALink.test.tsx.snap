// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CTALink > renders primary variant as expected 1`] = `
<div>
  <a
    class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base cta-button--primary bg-bluedot-normal text-white hover:bg-bluedot-normal px-4 py-2"
    href="https://example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="cta-button__text"
    >
      Click me
    </span>
  </a>
</div>
`;

exports[`CTALink > renders secondary variant as expected 1`] = `
<div>
  <a
    class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base cta-button--secondary bg-transparent text-bluedot-normal border border-bluedot-normal hover:bg-bluedot-lighter px-4 py-2"
    href="https://example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="cta-button__text"
    >
      Click me
    </span>
  </a>
</div>
`;

exports[`CTALink > renders with chevron as expected 1`] = `
<div>
  <a
    class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base cta-button--primary bg-bluedot-normal text-white hover:bg-bluedot-normal px-4 py-2"
    href="https://example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="cta-button__text"
    >
      Click me
    </span>
    <span
      class="cta-button__chevron ml-3"
    >
      <img
        alt="â†’"
        class="cta-button__chevron-icon w-2 h-2"
        src="/icons/chevron_white.svg"
      />
    </span>
  </a>
</div>
`;

exports[`CTALink > renders with custom className 1`] = `
<div>
  <a
    class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base cta-button--primary bg-bluedot-normal text-white hover:bg-bluedot-normal px-4 py-2 custom-class"
    href="https://example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="cta-button__text"
    >
      Click me
    </span>
  </a>
</div>
`;
